src/components/SidebarState.vue

```diff
@@ -6,11 +6,20 @@ import Vue from 'vue';
 export default Vue.extend({
     data() {
         return {
-            sidebarOpen: false,
-            // sidebarSection may be either '', 'user', 'settings', 'nicklist', 'about'
-            sidebarSection: '',
-            sidebarUser: null,
             activeComponent: null,
+            // sidebarSection may be either '', 'user', 'settings', 'nicklist', 'about'
+            channelState: {
+                allowedSections: ['user', 'settings', 'nicklist', 'about'],
+                sidebarOpen: false,
+                sidebarUser: null,
+                sidebarSection: '',
+            },
+            queryState: {
+                allowedSections: ['user'],
+                sidebarOpen: null,
```

Comment:
Why is this null instead of a bool? I see some other places that compares this with null too for some reason

Reply:
This will cause the sidebar to open in queries if it was open in channel when the first query is opened